src/nvim/ui.coffee

```diff
@@ -104,6 +106,34 @@ class UI extends EventEmitter
 
         , 250
 
+    @canvas.addEventListener 'mousewheel', (e) =>
+      {wheelDeltaX, wheelDeltaY} = event
+
+      @mouseDeltaX += wheelDeltaX
+      @mouseDeltaY += wheelDeltaY
+
+      mouseAbsX = Math.abs(Math.round(@mouseDeltaX * config.scroll_sensitivity))
+      mouseAbsY = Math.abs(Math.round(@mouseDeltaY * config.scroll_sensitivity))
+
+      if mouseAbsX > 0
+        direction = if @mouseDeltaX > 0 then 'Left' else 'Right'
+        button = get_vim_button_name('ScrollWheel' + direction, e)
+
+        for i in [1..mouseAbsX]
+          @emit 'input', button
```

Comment:
Neovim supports scroll event with details on both axes at the same time. Please refer to the gtk-ui in python-client.